common/rs-config.cpp

```diff
@@ -29,30 +25,30 @@ void config_file::set_default(const char* key, const char* calculate)
 
 void config_file::remove(const char* key)
 {
-    _values.erase(key);
+    _j.erase(key);
     save();
 }
 
 void config_file::reset()
 {
-    _values.clear();
+    _j = nlohmann::json::object();
     save();
 }
 
 std::string config_file::get(const char* key, const char* def) const
 {
-    auto it = _values.find(key);
-    if (it != _values.end())
+    auto it = _j.find(key);
+    if (it != _j.end() && it->is_string())
     {
-        return it->second;
+        return rsutils::json::string_ref( *it );
     }
     return get_default(key, def);
 }
 
 bool config_file::contains(const char* key) const
 {
-    auto it = _values.find(key);
-    return it != _values.end();
+    auto it = _j.find(key);
+    return it != _j.end() && it->is_string();
```

Comment:
Why check for string? Don't we support other values?

Reply:
Because, at least in rs-config, and in this PR, I didn't want to change any existing functionality: tools will keep behaving the same.

Reply:
My main goal was to make rs-config **compatible** with raw json -- before, it couldn't even load -- and able to keep non-string content intact.

Reply:
So in the next PR this is about to change?

Reply:
No, I'm not touching it again without a good reason.